Crash when gotify isn't available

**Describe the bug**

Crash of watchtower because gotify isn't available anymore because gotify or the reverse proxy got taken down for an update. Because of the crash my reverse proxy doesn't get restarted :/

**To Reproduce**
Steps to reproduce the behavior:
1. Setup gotify notifications
2. Start watchtower
3. Wait until gotify or the reverse proxy get taken down by watchdower for an update
4. See the crash log

**Expected behavior**

Watchtower tries to resend the notifications at a later point in time or just simply discards them.

**Screenshots**
If applicable, add screenshots to help explain your problem.

**Environment**
 - Platform: Docker
 - Architecture: amd64
 - Docker version: 19.03.6, build 369ce74a3c

**Logs from running watchtower**

```
time="2020-02-24T19:39:48+01:00" level=info msg="Starting Watchtower and scheduling first run: 2020-02-24 19:44:48 +0100 CET m=+300.695743863"
time="2020-02-24T19:44:59+01:00" level=info msg="Found new linuxserver/letsencrypt:latest image (sha256:c439abf13c1b55fc8b6f72e3e2038ce9f4d4406af9e051e9208fd5b83d440ecb)"
time="2020-02-24T19:45:16+01:00" level=info msg="Stopping /letsencrypt (a73d4a54c090549edd39d5c5a133b15b0d15da167c023c83fd458f1b55793739) with SIGTERM"
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x9b5ed1]

goroutine 27 [running]:
github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire.func1(0xc00012c1c0, 0xc00029cf00)
	/home/circleci/repo/pkg/notifications/gotify.go:87 +0x3f1
created by github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire
	/home/circleci/repo/pkg/notifications/gotify.go:71 +0x49
Gotify notification returned 502 HTTP status codeGotify notification returned 502 HTTP status codeGotify notification returned 502 HTTP status codeGotify notification returned 502 HTTP status codeGotify notification returned 502 HTTP status codeGotify notification returned 502 HTTP status codeGotify notification returned 502 HTTP status codeFailed to send Gotify notification:  Post https://gotify.domain.tld/message?token=WATCHTOWER_NOTIFICATION_GOTIFY_TOKEN: dial tcp <IP>:443: connect: connection refused
time="2020-02-24T19:45:40+01:00" level=info msg="Starting Watchtower and scheduling first run: 2020-02-24 19:50:40 +0100 CET m=+300.254307423"
Failed to send Gotify notification:  Post https://gotify.domain.tld/message?token=WATCHTOWER_NOTIFICATION_GOTIFY_TOKEN: dial tcp <IP>:443: connect: connection refused
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x9b5ed1]

goroutine 8 [running]:
github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire.func1(0xc0003583f0, 0xc0002a2ec0)
	/home/circleci/repo/pkg/notifications/gotify.go:87 +0x3f1
created by github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire
	/home/circleci/repo/pkg/notifications/gotify.go:71 +0x49
time="2020-02-24T19:45:44+01:00" level=info msg="Starting Watchtower and scheduling first run: 2020-02-24 19:50:44 +0100 CET m=+300.876220453"
Failed to send Gotify notification:  Post https://gotify.domain.tld/message?token=WATCHTOWER_NOTIFICATION_GOTIFY_TOKEN: dial tcp <IP>:443: connect: connection refused
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x9b5ed1]

goroutine 39 [running]:
github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire.func1(0xc000122540, 0xc000068880)
	/home/circleci/repo/pkg/notifications/gotify.go:87 +0x3f1
created by github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire
	/home/circleci/repo/pkg/notifications/gotify.go:71 +0x49
time="2020-02-24T19:45:53+01:00" level=info msg="Starting Watchtower and scheduling first run: 2020-02-24 19:50:53 +0100 CET m=+300.650850234"
Failed to send Gotify notification:  Post https://gotify.domain.tld/message?token=WATCHTOWER_NOTIFICATION_GOTIFY_TOKEN: dial tcp <IP>:443: connect: connection refused
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x9b5ed1]

goroutine 22 [running]:
github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire.func1(0xc0000aa690, 0xc000296ec0)
	/home/circleci/repo/pkg/notifications/gotify.go:87 +0x3f1
created by github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire
	/home/circleci/repo/pkg/notifications/gotify.go:71 +0x49
time="2020-02-24T19:45:58+01:00" level=info msg="Starting Watchtower and scheduling first run: 2020-02-24 19:50:58 +0100 CET m=+300.847314302"
Failed to send Gotify notification:  Post https://gotify.domain.tld/message?token=WATCHTOWER_NOTIFICATION_GOTIFY_TOKEN: dial tcp <IP>:443: connect: connection refused
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x9b5ed1]

goroutine 12 [running]:
github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire.func1(0xc00008e460, 0xc00031e800)
	/home/circleci/repo/pkg/notifications/gotify.go:87 +0x3f1
created by github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire
	/home/circleci/repo/pkg/notifications/gotify.go:71 +0x49
time="2020-02-24T19:46:03+01:00" level=info msg="Starting Watchtower and scheduling first run: 2020-02-24 19:51:03 +0100 CET m=+300.999705086"
panic: runtime error: invalid memory address or nil pointer dereference
Failed to send Gotify notification:  Post https://gotify.domain.tld/message?token=WATCHTOWER_NOTIFICATION_GOTIFY_TOKEN: dial tcp <IP>:443: connect: connection refused
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x9b5ed1]

goroutine 24 [running]:
github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire.func1(0xc00012e310, 0xc0002a4ec0)
	/home/circleci/repo/pkg/notifications/gotify.go:87 +0x3f1
created by github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire
	/home/circleci/repo/pkg/notifications/gotify.go:71 +0x49
time="2020-02-24T19:46:11+01:00" level=info msg="Starting Watchtower and scheduling first run: 2020-02-24 19:51:11 +0100 CET m=+300.963241131"
Failed to send Gotify notification:  Post https://gotify.domain.tld/message?token=WATCHTOWER_NOTIFICATION_GOTIFY_TOKEN: dial tcp <IP>:443: connect: connection refused
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x9b5ed1]

goroutine 8 [running]:
github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire.func1(0xc00012e460, 0xc00007c840)
	/home/circleci/repo/pkg/notifications/gotify.go:87 +0x3f1
created by github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire
	/home/circleci/repo/pkg/notifications/gotify.go:71 +0x49
time="2020-02-24T19:46:20+01:00" level=info msg="Starting Watchtower and scheduling first run: 2020-02-24 19:51:20 +0100 CET m=+300.503381333"
Failed to send Gotify notification:  Post https://gotify.domain.tld/message?token=WATCHTOWER_NOTIFICATION_GOTIFY_TOKEN: dial tcp <IP>:443: connect: connection refused
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x9b5ed1]

goroutine 10 [running]:
github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire.func1(0xc00012e3f0, 0xc0002a2ec0)
	/home/circleci/repo/pkg/notifications/gotify.go:87 +0x3f1
created by github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire
	/home/circleci/repo/pkg/notifications/gotify.go:71 +0x49
time="2020-02-24T19:46:36+01:00" level=info msg="Starting Watchtower and scheduling first run: 2020-02-24 19:51:36 +0100 CET m=+300.455139053"
panic: runtime error: invalid memory address or nil pointer dereference
Failed to send Gotify notification:  Post https://gotify.domain.tld/message?token=WATCHTOWER_NOTIFICATION_GOTIFY_TOKEN: dial tcp <IP>:443: connect: connection refused
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x9b5ed1]

goroutine 6 [running]:
github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire.func1(0xc0000aa4d0, 0xc000068840)
	/home/circleci/repo/pkg/notifications/gotify.go:87 +0x3f1
created by github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire
	/home/circleci/repo/pkg/notifications/gotify.go:71 +0x49
time="2020-02-24T19:47:07+01:00" level=info msg="Starting Watchtower and scheduling first run: 2020-02-24 19:52:07 +0100 CET m=+300.527729461"
Failed to send Gotify notification:  Post https://gotify.domain.tld/message?token=WATCHTOWER_NOTIFICATION_GOTIFY_TOKEN: dial tcp <IP>:443: connect: connection refused
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x9b5ed1]

goroutine 9 [running]:
github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire.func1(0xc000126380, 0xc000068840)
	/home/circleci/repo/pkg/notifications/gotify.go:87 +0x3f1
created by github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire
	/home/circleci/repo/pkg/notifications/gotify.go:71 +0x49
time="2020-02-24T19:48:04+01:00" level=info msg="Starting Watchtower and scheduling first run: 2020-02-24 19:53:04 +0100 CET m=+301.048134421"
Failed to send Gotify notification:  Post https://gotify.domain.tld/message?token=WATCHTOWER_NOTIFICATION_GOTIFY_TOKEN: dial tcp <IP>:443: connect: connection refused
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x9b5ed1]

goroutine 37 [running]:
github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire.func1(0xc00012e2a0, 0xc00029ef00)
	/home/circleci/repo/pkg/notifications/gotify.go:87 +0x3f1
created by github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire
	/home/circleci/repo/pkg/notifications/gotify.go:71 +0x49
time="2020-02-24T19:49:09+01:00" level=info msg="Starting Watchtower and scheduling first run: 2020-02-24 19:54:09 +0100 CET m=+300.468303304"
Failed to send Gotify notification:  Post https://gotify.domain.tld/message?token=WATCHTOWER_NOTIFICATION_GOTIFY_TOKEN: dial tcp <IP>:443: connect: connection refused
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x9b5ed1]

goroutine 35 [running]:
github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire.func1(0xc0001224d0, 0xc000068840)
	/home/circleci/repo/pkg/notifications/gotify.go:87 +0x3f1
created by github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire
	/home/circleci/repo/pkg/notifications/gotify.go:71 +0x49
time="2020-02-24T19:50:14+01:00" level=info msg="Starting Watchtower and scheduling first run: 2020-02-24 19:55:14 +0100 CET m=+300.545620758"
Failed to send Gotify notification:  Post https://gotify.domain.tld/message?token=WATCHTOWER_NOTIFICATION_GOTIFY_TOKEN: dial tcp <IP>:443: connect: connection refused
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x9b5ed1]

goroutine 8 [running]:
github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire.func1(0xc000122540, 0xc000068880)
	/home/circleci/repo/pkg/notifications/gotify.go:87 +0x3f1
created by github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire
	/home/circleci/repo/pkg/notifications/gotify.go:71 +0x49
time="2020-02-24T19:51:19+01:00" level=info msg="Starting Watchtower and scheduling first run: 2020-02-24 19:56:19 +0100 CET m=+300.309542440"
Failed to send Gotify notification:  Post https://gotify.domain.tld/message?token=WATCHTOWER_NOTIFICATION_GOTIFY_TOKEN: dial tcp <IP>:443: connect: connection refused
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x9b5ed1]

goroutine 39 [running]:
github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire.func1(0xc0000aa5b0, 0xc0002a2f00)
	/home/circleci/repo/pkg/notifications/gotify.go:87 +0x3f1
created by github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire
	/home/circleci/repo/pkg/notifications/gotify.go:71 +0x49
time="2020-02-24T19:52:24+01:00" level=info msg="Starting Watchtower and scheduling first run: 2020-02-24 19:57:24 +0100 CET m=+300.514561245"
panic: runtime error: invalid memory address or nil pointer dereference
Failed to send Gotify notification:  Post https://gotify.domain.tld/message?token=WATCHTOWER_NOTIFICATION_GOTIFY_TOKEN: dial tcp <IP>:443: connect: connection refused
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x9b5ed1]

goroutine 6 [running]:
github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire.func1(0xc000122540, 0xc000068840)
	/home/circleci/repo/pkg/notifications/gotify.go:87 +0x3f1
created by github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire
	/home/circleci/repo/pkg/notifications/gotify.go:71 +0x49
time="2020-02-24T19:53:30+01:00" level=info msg="Starting Watchtower and scheduling first run: 2020-02-24 19:58:30 +0100 CET m=+300.212907260"
Failed to send Gotify notification:  Post https://gotify.domain.tld/message?token=WATCHTOWER_NOTIFICATION_GOTIFY_TOKEN: dial tcp <IP>:443: connect: connection refused
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x9b5ed1]

goroutine 23 [running]:
github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire.func1(0xc00010c2a0, 0xc000320800)
	/home/circleci/repo/pkg/notifications/gotify.go:87 +0x3f1
created by github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire
	/home/circleci/repo/pkg/notifications/gotify.go:71 +0x49
time="2020-02-24T19:54:35+01:00" level=info msg="Starting Watchtower and scheduling first run: 2020-02-24 19:59:35 +0100 CET m=+301.019653552"
Failed to send Gotify notification:  Post https://gotify.domain.tld/message?token=WATCHTOWER_NOTIFICATION_GOTIFY_TOKEN: dial tcp <IP>:443: connect: connection refused
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x9b5ed1]

goroutine 58 [running]:
github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire.func1(0xc00012e460, 0xc0002a2ec0)
	/home/circleci/repo/pkg/notifications/gotify.go:87 +0x3f1
created by github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire
	/home/circleci/repo/pkg/notifications/gotify.go:71 +0x49
time="2020-02-24T19:55:44+01:00" level=info msg="Starting Watchtower and scheduling first run: 2020-02-24 20:00:44 +0100 CET m=+300.807764954"
Failed to send Gotify notification:  Post https://gotify.domain.tld/message?token=WATCHTOWER_NOTIFICATION_GOTIFY_TOKEN: dial tcp <IP>:443: connect: connection refused
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x9b5ed1]

goroutine 8 [running]:
github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire.func1(0xc0001224d0, 0xc00007c840)
	/home/circleci/repo/pkg/notifications/gotify.go:87 +0x3f1
created by github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire
	/home/circleci/repo/pkg/notifications/gotify.go:71 +0x49

```

Shortened Log with debug Info:

```
time="2020-02-25T05:10:19+01:00" level=info msg="Stopping /letsencrypt (cc24d816e39b508debe258dfa49fb149273904c516db7bae0b8764c2268cf369) with SIGTERM"
time="2020-02-25T05:10:27+01:00" level=debug msg="Removing container cc24d816e39b508debe258dfa49fb149273904c516db7bae0b8764c2268cf369"
time="2020-02-25T05:10:33+01:00" level=debug msg="This is the watchtower container /watchtower"
time="2020-02-25T05:10:33+01:00" level=info msg="Creating /letsencrypt"
Gotify notification returned 401 HTTP status codeGotify notification returned 401 HTTP status codeGotify notification returned 401 HTTP status codeFailed to send Gotify notification:  Post https://gotify.domain.tld/message?token=WATCHTOWER_NOTIFICATION_GOTIFY_TOKEN: dial tcp <IP>:443: connect: connection refused
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x9b5ed1]

goroutine 136 [running]:
github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire.func1(0xc0000a43f0, 0xc000280f00)
        /home/circleci/repo/pkg/notifications/gotify.go:87 +0x3f1
created by github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire
        /home/circleci/repo/pkg/notifications/gotify.go:71 +0x49
time="2020-02-25T05:10:36+01:00" level=debug msg="Sleeping for a second to ensure the docker api client has been properly initialized."
time="2020-02-25T05:10:37+01:00" level=debug msg="Retrieving running containers"
time="2020-02-25T05:10:37+01:00" level=debug msg="There are no additional watchtower containers"
time="2020-02-25T05:10:37+01:00" level=info msg="Starting Watchtower and scheduling first run: 2020-02-25 05:15:37 +0100 CET m=+300.273456239"
Failed to send Gotify notification:  Post https://gotify.domain.tld/message?token=WATCHTOWER_NOTIFICATION_GOTIFY_TOKEN: dial tcp <IP>:443: connect: connection refused
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x9b5ed1]

goroutine 9 [running]:
github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire.func1(0xc000338460, 0xc000296ec0)
        /home/circleci/repo/pkg/notifications/gotify.go:87 +0x3f1
created by github.com/containrrr/watchtower/pkg/notifications.(*gotifyTypeNotifier).Fire
        /home/circleci/repo/pkg/notifications/gotify.go:71 +0x49
time="2020-02-25T05:10:40+01:00" level=debug msg="Sleeping for a second to ensure the docker api client has been properly initialized."
time="2020-02-25T05:10:41+01:00" level=debug msg="Retrieving running containers"
time="2020-02-25T05:10:41+01:00" level=debug msg="There are no additional watchtower containers"
time="2020-02-25T05:10:41+01:00" level=info msg="Starting Watchtower and scheduling first run: 2020-02-25 05:15:41 +0100 CET m=+300.873903243"
Failed to send Gotify notification:  Post https://gotify.domain.tld/message?token=WATCHTOWER_NOTIFICATION_GOTIFY_TOKEN: dial tcp <IP>:443: connect: connection refused
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x40 pc=0x9b5ed1]
```

EDIT: Added shortened debug log

--- a/pkg/notifications/gotify.go
+++ b/pkg/notifications/gotify.go
@@ -83,6 +83,7 @@ func (n *gotifyTypeNotifier) Fire(entry *log.Entry) error {
 		resp, err := http.Post(n.getURL(), "application/json", jsonBodyBuffer)
 		if err != nil {
 			fmt.Println("Failed to send Gotify notification: ", err)
+			return
 		}
 		defer resp.Body.Close()
 

